solver/src/oneinch_solver.rs

```diff
@@ -264,7 +265,7 @@ mod tests {
     #[tokio::test]
     #[ignore]
     async fn returns_error_on_non_mainnet() {
-        let web3 = testutil::infura("rinkeby");
+        let web3 = Web3::new(create_test_transport(&std::env::var("NODE_URL").unwrap()));
```

Comment:
This test specifically needs to connect to a non-mainnet node in order to pass.

Reply:
I understand that but imo you should set NODE_URL appropriately as you do for all the other tests.

Reply:
But `testutil::infura()` already requires setting `INFURA_PROJECT_ID` implying that you are configuring the tests for use with Infura. My only concern with using `NODE_URL` here is that it makes it impossible to run both `solve_order_on_oneinch` and `returns_error_on_non_mainnet` in a single run.

Reply:
On the other hand I can't run the tests if I'm not using infura.

Reply:
Hmm, good point... What I don't like though is that `cargo test -p solver -- --ignored oneinch` to run all 1Inch solver tests (including the ignored ones) can never succeed. That being said, requiring Infura configuration for the test is also quite annoying.

Maybe `env::var::RINKEBY_NODE_URL`?